jsonschema-core.xml

```diff
@@ -455,6 +455,68 @@
 
         </section>
 
+        <section title="Principles">
+            <t>
+                JSON Schema is built on several principles, which both explain various design
+                choices, and provide guidance on how to extend the specification effectively.
+            </t>
+            <!-- Links will be added to relevant spec parts after the sections are reorganized
+                 to create the new schema loading section -->
+            <t>
+                <list>
+                    <t>
+                        All syntactically legal JSON documents with well-defined behavior are fully
+                        supported as instance data.  Notably, JSON objects with duplicate property
+                        names are syntactically legal but not well-defined.
+                    </t>
+                    <t>
+                        JSON Schema is defined over a data model, not JSON text.  Therefore
+                        JSON Schema can be used with any data format that can be reliably parsed
+                        into the data model.
+                    </t>
+                    <t>
+                        Locations in schemas and instances are always identified by URIs,
+                        JSON Pointers, or Relative JSON Pointers.  JSON Schema treats URIs as
+                        identifiers only.  Locating and retrieving URI-identified resources
+                        is outside of the scope of this specification.
+                    </t>
+                    <t>
+                        Schema URIs referenced or defined in JSON Schema's Core vocabulary MUST be
+                        resolvable without instance data.
+                    </t>
+                    <t>
+                        Once schema URIs are resolved, the result of applying a schema object
+                        to instance data is data is a function of that schema object and its
+                        dynamic-scope subschemas.  Parent schemas have no effect on results.
+                    </t>
+                    <t>
+                        Each keyword's behavior falls into one or more behavioral classifications.
+                        A keyword's classification MUST be the same for all values of the keyword.
+                    </t>
+                    <t>
+                        Keywords produce a boolean assertion result, and an annotation result that
+                        can be of any type.  Depending on the classification, the assertion result
+                        might always be true, or the annotation result might always be empty.
```

Comment:
Annotations are keywords and they do not produce any assertion result, right?

Reply:
This defines that *all* keywords produce both, but "annotation" keywords always produce a `true` assertion, making the assertion result moot.

Reply:
@gregsdennis that is correct.  Might need to also clarify what "empty" annotation results mean.

I put this in because in terms of generic extensions, just always returning the same sorts of things makes sense instead of "wait, what kind of keyword was this?  does it return this kind of thing? yes? no?"  If not otherwise specified, keywords return a `true` validation result.

This is open to debate, it just made the most sense as I was typing it.

(this is exactly why laying out principles is a good idea, though! I don't think we've ever really discussed this)

Reply:
OK. Fair.

Reply:
Looking at the document, I can't see this defined explicitly. Are you saying it's implicit with the above wording? If so, could we ALSO make it explicit with requirement keywords? Tests will then have a requirements keyword to point to in the spec then.

Reply:
@Relequestual I think we should make it explicit somewhere else.  I specifically don't want any requirement keywords in the Principles section:  They're principles, not requirements.  They should drive the requirements.  I will make a separate PR for clarifying the nature of results.

Reply:
(I'm taking the MAY out of one of the other principles and changing it to "can")